### PR TITLE
After building project

### PR DESCRIPTION
**Pull request by Locutus18**
Fixes : #74 
It was created with the following rules: if a file is open, the bugs shown here are the bugs in the project from which the file is open; if there isn't one, the bugs of the most recently built project.

### CODE REVIEW

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 1]
```diff
@@ -101,15 +101,14 @@ private void refreshCustom(IWorkbenchPage[] pages, IProject project, String seco
             for (IViewReference vp : page.getViewReferences()) {
                 if (vp.getId().equals(ReportListViewCustom.ID)) {
                     ReportListViewCustom rlvc = (ReportListViewCustom) vp.getView(true);
-                    if(secondaryId.equals("")) {
+                    if(secondaryId.equals("") && rlvc.getViewSite().getSecondaryId() != null) {
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: Please add javadoc comment for the function headers.

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 2]
```diff
@@ -47,17 +47,32 @@
...
 public class CodeCheckerContext {
 
-    //Logger
+    /** The Constant logger. */
     private static final Logger logger = LogManager.getLogger(CodeCheckerContext.class);
 
+    /** The instance. */
     static CodeCheckerContext instance;
+
+    /** The job runner. */
```
Comment (gyorb) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: For what type of jobs do we use the job runner?

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 3]
```diff
@@ -47,17 +47,32 @@
...
     JobRunner jobRunner;
+
+    /** The active editor part. */
```
Comment (gyorb) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: What can be an active editor part?

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 4]
```diff
@@ -66,6 +81,13 @@ private CodeCheckerContext() {
         jobRunner = new SimpleJobRunner(actionRunner);
     }
 
+    /**
+     * The refresher for Project ReportList View.
```
Comment (gyorb) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: Does the refreshing mean cleaning the cached values and get the results again from the CodeChecker server?

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 5]
```diff
@@ -267,6 +341,13 @@ public void refreshAddCustomReportListView(String secondaryId) {
         this.activeProject = project;
     }
 
+    /**
+     * Run report job.
+     *
+     * @param target the target
```
Comment (gyorb) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: I'm not sure these parameter comments helps to understand the method.

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 6]
```diff
@@ -66,6 +81,13 @@ private CodeCheckerContext() {
         jobRunner = new SimpleJobRunner(actionRunner);
     }
 
+    /**
+     * The refresher for Project ReportList View.
+     * 
+     * @param pages the page list for the currently active workbench windows.
+     * @param project the project which the user change there view to.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: -> the project to change the view to

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 7]
```diff
@@ -66,6 +81,13 @@ private CodeCheckerContext() {
...
+     * @param considerViewerRefresh false if the refresh should always happen despite of no real need to refresh
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: -> force refresh

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java (hunk 4 above)
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: Locotus18, please extend the description of the function to answer this question.

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 8]
```diff
@@ -95,34 +125,56 @@ private void refreshCurrent(IWorkbenchPage[] pages, IProject project, String fil
...
     }
 
+    /**
+     * The refresher for Custom ReportList View. If secondary id is empty, 
+     * it checks if a refresh really needs to happen and if so updates every 
+     * custom view for the current project. If secondary-id is not empty, 
+     * it will search for the gives secondary-id custom view and 
+     * updates that particular one.
+     *
+     * @param pages the page list for the currently active workbench windows.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: -> currently active workbench window.
there can be only one active window, right?

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 9]
```diff
@@ -146,12 +198,21 @@ public void run() {
         return servers.get(project);
     }
 
+    /**
+     * Clean cache.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: what is this cache caching. please extend the description.

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 10]
```diff
@@ -228,6 +293,11 @@ public void refreshChangeEditorPart(IEditorPart partRef) {
         this.activeProject = project;
     }
 
+    /**
+     * Refresh change project.
+     *
+     * @param project the project which the user change there view to.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: "change there view to"
-> project to change change the view to.

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 11]
```diff
@@ -243,6 +313,11 @@ public void refreshChangeProject(IProject project) {
         this.activeProject = project;
     }
 
+    /**
+     * Refresh add custom report list view.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: what does this function add?

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 12]
```diff
@@ -266,6 +341,13 @@ public void refreshAddCustomReportListView(String secondaryId) {
         this.activeProject = project;
     }
 
+    /**
+     * Run report job.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: what is a report job? please explain

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 13]
```diff
@@ -280,6 +362,11 @@ public void runReportJob(ReportListView target, ImmutableList<ResultFilter> filt
         jobRunner.addJob(rlj);
     }
 
+    /**
+     * Run run list job.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: what is a list job?

File: eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java [hunk 14]
```diff
@@ -310,6 +397,11 @@ public void onJobComplete(RunListJob rlj) {
         jobRunner.addJob(rlj);
     }
 
+    /**
+     * Run analyze job.
```
Comment (dkrupp) [inline eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CodeCheckerContext.java]: what is an analyze job?

**Pull request merged**